### PR TITLE
[DET-2039] Decoupling code climate from coverage

### PR DESCRIPTION
**Pull request by dasphillipbrau**
**What this PR does / why we need it**:

**Special notes for your reviewer**:

**If applicable**:
- [ ] All new Jobs, Commands, Executors, Parameters have descriptions
- [ ] If PR adds a new feature, Examples have been added
- [ ] README has been updated, if necessary

### CODE REVIEW

**Review by jobegrabber — CHANGES_REQUESTED** [review]
Please put [these steps](https://github.com/travelaudience/orbs/pull/136/files#diff-89883f4ceb7c4b814a912a52ff6ded8afdbbe9a2b9588b44001bcc63c7462231R41-R44) under the condition, too.

[These steps](https://github.com/travelaudience/orbs/pull/136/files#diff-89883f4ceb7c4b814a912a52ff6ded8afdbbe9a2b9588b44001bcc63c7462231R80-L86) we also want to run without CodeClimate

File: src/scala/jobs/run_tests.yaml [hunk 1]
```diff
@@ -97,6 +100,11 @@ steps:
                 echo 'export TESTS_STATUS=1' >> $BASH_ENV
               fi
 
+  # Clean all reports to reduce the cache size
+  - run:
```
[the remark this answers is not in the document]
Comment (dasphillipbrau) [reply src/scala/jobs/run_tests.yaml:104]: Good catch

File: src/scala/commands/test.yaml [hunk 2]
```diff
@@ -35,7 +40,18 @@ parameters:
 
 steps:
   - when:
-      condition: << parameters.with_coverage >>
+      condition:
+        and:
+        - equal: [ true, << parameters.with_code_climate >>]
+        - equal: [ false, << parameters.with_coverage >>]
```
Comment (jobegrabber) [inline src/scala/commands/test.yaml:46]: Could you try this?

```suggestion
        - << parameters.with_code_climate >>]
        - not: << parameters.with_coverage >>
```

File: src/scala/commands/test.yaml [hunk 3]
```diff
@@ -78,6 +88,15 @@ steps:
...
             destination: scoverage-report
+        - when:
+            condition: << parameters.with_code_climate >>
+            steps:
+              - code-climate/format_coverage:
+                  cc_test_reporter_path: /tmp/code-climate
+                  coverage_file: << parameters.project_base >>/<< parameters.path_to_coverage_file >>
+                  input_type: cobertura
+              - code-climate/upload_coverage:
+                  cc_test_reporter_path: /tmp/code-climate
```
Comment (jobegrabber) [inline src/scala/commands/test.yaml:99]: these are nested steps, right? interesting that it seems to work :)

**Review by jobegrabber — APPROVED** [review]

**Pull request merged**